pkg/kubelet/volume_host.go

```diff
@@ -278,6 +279,18 @@ func (kvh *kubeletVolumeHost) GetNodeLabels() (map[string]string, error) {
 	return node.Labels, nil
 }
 
+func (kvh *kubeletVolumeHost) GetNodeFields() (map[string]string, error) {
+	node, err := kvh.kubelet.GetNode()
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving node: %v", err)
+	}
+	nodeFields := map[string]string{}
```

Comment:
For reviewers: this code is now duplicated a couple of times across the codebase. It could make sense to factor it out, but then where should such utility function live? It should have access to `NodeFieldSelectorKeys` as well, so may make sense to keep 'closeby'.

Reply:
What if we just have a new `GetNode()` method, and remove `GetNodeLabels()`? That way we don't have to have 2 `CheckNodeAffinity()` methods